controllers/applicationsnapshotenvironmentbinding_controller.go

```diff
@@ -176,6 +179,46 @@ func (r *SnapshotEnvironmentBindingReconciler) Reconcile(ctx context.Context, re
 			return ctrl.Result{}, err
 		}
 
+		var clusterIngressDomain string
+		isKubernetesCluster := isKubernetesCluster(environment)
+		unsupportedConfig := environment.Spec.UnstableConfigurationFields
+		if unsupportedConfig != nil {
+			clusterIngressDomain = unsupportedConfig.IngressDomain
+		}
+
+		// Safeguard if Ingress Domain is empty on Kubernetes
+		if isKubernetesCluster && clusterIngressDomain == "" {
+			err = fmt.Errorf("ingress domain cannot be empty on a Kubernetes cluster")
+			log.Error(err, "unable to create an ingress resource on a Kubernetes cluster")
+			r.SetConditionAndUpdateCR(ctx, req, &appSnapshotEnvBinding, err)
+			return ctrl.Result{}, err
+		}
+
+		devfileSrc := devfile.DevfileSrc{
+			Data: hasComponent.Status.Devfile,
+		}
+		compDevfileData, err := devfile.ParseDevfile(devfileSrc)
+		if err != nil {
+			errMsg := fmt.Sprintf("Unable to parse the devfile from Component status, exiting reconcile loop %v", req.NamespacedName)
+			log.Error(err, errMsg)
+			r.SetConditionAndUpdateCR(ctx, req, &appSnapshotEnvBinding, fmt.Errorf("%v: %v", errMsg, err))
+			return ctrl.Result{}, err
+		}
+
+		deployAssociatedComponents, err := devfileParser.GetDeployComponents(compDevfileData)
```

Comment:
No, so this is how this branch works:

- if devfile.kubernetes.endpoint is mentioned we create an ingress/route from it and pass it to gitops gen overlays
- if devfile.kubernetes.endpoint is not mentioned, then 2 things can happen:
      - we use the ingress/route mentioned in the outerloop yaml file
      - if no ingress/route is present in the outerloop yaml file, then gitops gen will generate one based on the info in the Component resource

Reply:
Summary of the prev post:

base dir -> deployment.yaml, service.yaml, kustomization.yaml, other-resources.yaml
overlays dir -> deployment-patch.yaml, kustomization.yaml, ingress.yaml/route.yaml

Possible resources in other-resources.yaml:
1. ingress1, ingress2, pvc1, pvc2 defined in devfile outerloop if devfile.kuberntes.endpoint is mentioned. devfile.kuberntes.endpoint goes over to overlays' ingress.yaml
2. ingress2, pvc1, pvc2 defined in devfile outerloop if devfile.kuberntes.endpoint is NOT mentioned. ingress 1 goes over to overlays' ingress.yaml
3. pvc1,pvc2 defined in devfile outerloop and if devfile.kuberntes.endpoint is NOT mentioned; gitops gen library's ingress goes over to overlays' ingress.yaml

Reply:
Re: 

> if devfile.kubernetes.endpoint is not mentioned, then 2 things can happen:
> - we use the ingress/route mentioned in the outerloop yaml file
> - if no ingress/route is present in the outerloop yaml file, then gitops gen overlays will generate one based on the info in the Component resource

I think if no endpoint is defined in the devfile, then we should look at the target port. If it's set, we generate an ingress based off of it. If no target port, then we have no endpoint information available to us, and we don't generate an ingress (or service either it would seem). We leave any user defined ingresses under `other-resources.yaml`

Reply:
we discussed this in a F2F meeting on Friday and came to a consensus that what we have in this PR is good for service preview and we can look to improve it iteratively later.